src/query/storage/promremote/storage_test.go

```diff
@@ -453,3 +491,87 @@ func getWriteRequest(promServer *promremotetest.TestPromServer) *prompb.WriteReq
 	}
 	return wq
 }
+
+func LoadTestPromRemoteStorage(t *testing.T, jitter bool, numTenants, numSeries, numSamples int) {
+	fakeProm := promremotetest.NewServer(t, jitter)
+	defer fakeProm.Close()
+	scope := tally.NewTestScope("test_scope", map[string]string{})
+	defer verifyMetrics(t, scope)
+	labelName := "test_tag_name"
+	labelValues := make([][]byte, numSeries)
+	for i := 0; i < numSeries; i++ {
+		labelValues[i] = []byte(fmt.Sprintf("test_tag_value_%d", i))
+	}
+	tenantRules := make([]TenantRule, numTenants)
+	for i := 0; i < numTenants; i++ {
+		filterValues, _ := filters.ValidateTagsFilter(fmt.Sprintf("%s:%s", labelName, labelValues[i%numSeries]))
+		filter, _ := filters.NewTagsFilter(filterValues, filters.Conjunction, filters.TagsFilterOptions{})
+		tenantRules[i] = TenantRule{
+			Tenant: fmt.Sprintf("tenant_%d", i),
+			Filter: filter,
+		}
+	}
+	promStorage, err := NewStorage(Options{
+		endpoints:     []EndpointOptions{{name: "testEndpoint", address: fakeProm.WriteAddr(), tenantHeader: "TENANT"}},
+		scope:         scope,
+		logger:        logger,
+		poolSize:      10,
+		queueSize:     numTenants * numSamples,
```

Comment:
@yuchen-db  decrease the queue size will result in discrepancy.

<img width="927" alt="Screenshot 2024-05-09 at 3 52 10 PM" src="https://github.com/databricks/m3/assets/96499497/028507af-81d6-48ba-9ae0-05ebaad27059">